src/Common/StringSearcher.h

```diff
@@ -660,18 +667,54 @@ class StringSearcher<true, ASCII> : public StringSearcherBase
         if (needle == needle_end)
             return haystack;
 
-        while (haystack < haystack_end)
-        {
 #ifdef __SSE4_1__
-            if (haystack + n <= haystack_end && pageSafe(haystack))
+        /// Here is the quick path when needle_size is 1. Compare the first and second characters if the needle_size >= 2.
```

Comment:
Let's make the comment simpler and more precise: `/// Fast path for single-character needles. Compare 16 characters of the haystack against the needle character at once.`

Reply:
Done.